nixos/modules/services/web-servers/apache-httpd/vhost-options.nix

```diff
@@ -47,10 +47,27 @@ in
       ];
       description = ''
         Listen addresses and ports for this virtual host.
-        <note><para>
+        <note>
+        <para>
           This option overrides <literal>addSSL</literal>, <literal>forceSSL</literal> and <literal>onlySSL</literal>.
-        </para></note>
+        </para>
+        <para>
+          If you only want to set the addresses manually and not the ports, take a look at <literal>listenAddresses</literal>.
+        </para>
+        </note>
+      '';
+    };
+
+    listenAddresses = mkOption {
+      type = with types; listOf str;
+
+      description = ''
+        Listen addresses for this virtual host.
+        Compared to <literal>listen</literal> this only sets the addreses
+        and the ports are choosen automatically.
       '';
+      default = [];
```

Comment:
```suggestion
      default = [ "*" ];
```

Reply:
I actually had a branch that did some cleanup like this, but never ended up pushing it... With this PR I was going for as much copy/paste of #131962 as possible, but you're right, better is better :smile: I'll take a look at that.